enterprise/app/trends/trends.tsx

```diff
@@ -66,14 +70,32 @@ export default class TrendsComponent extends React.Component<Props> {
   }
 
   fetchStats() {
+    // TODO(bduffany): Cancel in-progress request
+
     let request = new invocation.GetTrendRequest();
-    request.lookbackWindowDays = this.getLimit();
     request.query = new invocation.TrendQuery();
 
-    if (this.state.filterOnlyCI) {
-      request.query.role = ["CI"];
+    if (capabilities.globalFilter) {
+      const filterParams = getProtoFilterParams(this.props.path, this.props.search);
+      if (filterParams.role) {
+        request.query.role = [filterParams.role];
+      } else {
+        // Note: Technically we're filtering out workflows and unknown roles,
+        // even though the user has selected "All roles". But we do this to
+        // avoid double-counting build times for workflows and their nested CI runs.
+        request.query.role = ["", "CI"];
```

Comment:
I think we should just let people pick if they want to see all or if they want to filter out workflows (using the new picker). I think we should avoid this kind of implicit behavior.